ui/components/app/transaction-decoding/components/decoding/address/address.component.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import copyToClipboard from 'copy-to-clipboard';
+import { getAccountLink } from '@metamask/etherscan-link';
+import { shortenAddress } from '../../../../../../helpers/utils/util';
+import Identicon from '../../../../../ui/identicon';
+import { useI18nContext } from '../../../../../../hooks/useI18nContext';
+import {
+  getAddressBook,
+  getCurrentChainId,
+  getRpcPrefsForCurrentProvider,
+} from '../../../../../../selectors';
+import NicknamePopover from '../../../../../ui/nickname-popover';
+import UpdateNicknamePopover from '../../../../../ui/update-nickname-popover';
+import { addToAddressBook } from '../../../../../../store/actions';
+
+const SHOW_NICKNAME_POPOVER = 'SHOW_NICKNAME_POPOVER';
+const ADD_NICKNAME_POPOVER = 'ADD_NICKNAME_POPOVER';
+
+const Address = ({
+  checksummedRecipientAddress,
+  onRecipientClick,
+  addressOnly,
+  recipientEns,
+  recipientName,
+}) => {
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+  const [addressCopied, setAddressCopied] = useState(false);
+  const [popoverToDisplay, setPopoverToDisplay] = useState(null);
+
+  const addressBook = useSelector(getAddressBook);
+  const chainId = useSelector(getCurrentChainId);
+  const addressBookEntryObject =
+    addressBook &&
+    addressBook[chainId] &&
+    addressBook[chainId][checksummedRecipientAddress];
+  const recipientNickname = addressBookEntryObject?.name;
+
+  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
+
+  const explorerLink = getAccountLink(
+    checksummedRecipientAddress,
+    chainId,
+    { blockExplorerUrl: rpcPrefs?.blockExplorerUrl ?? null },
+    null,
+  );
+
+  let tooltipHtml = <p>{t('copiedExclamation')}</p>;
+  if (!addressCopied) {
+    tooltipHtml = '';
+  }
```

Comment:
This `tooltipHtml` doesn't get used for anything - should it?

Reply:
you are right, I just removed it !